src/xcode/ENA/ENA/Source/AppDelegate+ENATaskExecutionDelegate.swift

```diff
@@ -71,44 +71,66 @@ extension AppDelegate: ENATaskExecutionDelegate {
 	/// This method performs a check for the current exposure detection state. Only if the risk level has changed compared to the
 	/// previous state, a local notification is shown.
 	private func executeExposureDetectionRequest(completion: @escaping ((Bool) -> Void)) {
+		Log.info("[ENATaskExecutionDelegate] Execute exposure detection.", log: .riskDetection)
 
 		// At this point we are already in background so it is safe to assume background mode is available.
 		riskProvider.riskProvidingConfiguration.detectionMode = .fromBackgroundStatus(.available)
 
-		riskProvider.requestRisk(userInitiated: false) { [weak self] result in
+		backgroundTaskConsumer = RiskConsumer()
```

Comment:
what happens is a backgroundTaskConsumer get overwritten here?
this will leak memory in riskProvider, the old backgroundTaskConsumer will never dealoc
also both consumers will receive notifications (didCalculateRisk)

Reply:
Good point! Removed line 79 since its already initialized.